src/Azure.Functions.Cli/Actions/AzureActions/BaseAzureAction.cs

```diff
@@ -89,10 +114,89 @@ private async Task<string> AzureCliGetToken()
+                if (await CommandChecker.PowerShellModuleExistsAsync(_windowsPowerShellExecutable, _azProfileModuleName))
+                {
+                    scriptToRun = GetPowerShellAccessTokenScript(_azProfileModuleName);
+                }
+                else if (await CommandChecker.PowerShellModuleExistsAsync(_windowsPowerShellExecutable, _azureRmProfileModuleName))
+                {
+                    scriptToRun = GetPowerShellAccessTokenScript(_azureRmProfileModuleName);
+                }
+                else
+                {
+                    throw new FileNotFoundException("Cannot find az cli or Azure PowerShell. Please make sure to install az cli or Azure PowerShell.");
+                }
+
+                var az = new Executable("powershell", $"-NonInteractive -o Text -NoProfile -c {scriptToRun}");
+
+                var stdout = new StringBuilder();
+                var stderr = new StringBuilder();
+                var exitCode = await az.RunAsync(o => stdout.AppendLine(o), e => stderr.AppendLine(e));
+                if (exitCode != 0)
+                {
+                    throw new CliException(stderr.ToString().Trim(' ', '\n', '\r') + $"{Environment.NewLine}" + "Make sure to run \"Login-AzAccount\" or \"Login-AzureRmAccount\" to log in to Azure and retry this command.");
+                }
+                else
+                {
+                    return stdout.ToString().Trim(' ', '\n', '\r', '"');
+                }
+            }
             else
             {
-                throw new FileNotFoundException("Cannot find az cli. Please make sure to install az cli.");
+                throw new FileNotFoundException("Cannot find az cli or Azure PowerShell. Please make sure to install az cli or Azure PowerShell.");
+            }
+        }
+
+        // Sets the prefix of the script in case they have Az.Profile or AzureRM.Profile
+        private static string GetPowerShellAccessTokenScript (string module)
+        {
+            string prefix;
+            if (module == _azProfileModuleName)
+            {
+                prefix = "Az";
             }
+            else if (module == _azureRmProfileModuleName)
+            {
+                prefix = "AzureRM";
+            }
+            else
+            {
+                throw new ArgumentException($"Expected module to be '{_azProfileModuleName}' or '{_azureRmProfileModuleName}'");
+            }
+
+            // This PowerShell script first grabs the Azure context, fetches the profile client and requests an accesstoken.
+            // This entirely done using the Az.Profile module or AzureRM.Profile
+            return $@"
+$currentAzureContext = Get-{prefix}Context;
+$azureRmProfile = [Microsoft.Azure.Commands.Common.Authentication.Abstractions.AzureRmProfileProvider]::Instance.Profile;
+$profileClient = New-Object Microsoft.Azure.Commands.ResourceManager.Common.RMProfileClient($azureRmProfile);
```

Comment:
Please use `New-Object <typename> $azureRmProfile` instead of `New-Object <typename>($azureRmProfile)`.

Reply:
fixed!